stack build error in linux docker container

Here's the issue I'm struggling with right now:
```bash
appuser@8be60e71ef81:~/user/halive$ stack build
Building all executables for `halive' once. After a successful build of all of them, only specified executables will be rebuilt.
halive-0.1.6: build (lib + exe)
Preprocessing library for halive-0.1.6..
Building library for halive-0.1.6..
[5 of 8] Compiling Halive.SubHalive ( src/Halive/SubHalive.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Halive/SubHalive.o )

/app/user/halive/src/Halive/SubHalive.hs:248:17: error: parse error on input ‘#’
    |
248 |                 #else
    |                 ^


--  While building package halive-0.1.6 using:
      /app/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-2.4.0.1 build lib:halive exe:halive --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

And my cpp :
```bash
appuser@8be60e71ef81:~/user/halive$ cpp --version
cpp (Ubuntu 7.3.0-16ubuntu3) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
appuser@8be60e71ef81:~/user/halive$ stack --version
Version 1.9.3, Git revision 40cf7b37526b86d1676da82167ea8758a854953b (6211 commits) x86_64 hpack-0.31.1
```

It makes absolutely no sense to me (I unerstand the preprocessor is not doing its job correctly, but is it the cpp preprocessor, or the ghc one ?) if somebody's got any direction I'll take it ...

Running `stack build halive --ghc-options="-fPIC -fasm"` still fails for me:
```
halive> configure
halive> Configuring halive-0.1.6...
halive> build
halive> Preprocessing library for halive-0.1.6..
halive> Building library for halive-0.1.6..
halive> [1 of 8] Compiling Halive.Args
halive> [2 of 8] Compiling Halive.Concurrent
halive> [3 of 8] Compiling Halive.FileListener
halive> [4 of 8] Compiling Halive.FindPackageDBs
halive> [5 of 8] Compiling Halive.SubHalive
halive> 
halive> /tmp/stack2801/halive-0.1.6/src/Halive/SubHalive.hs:248:17: error: parse error on input ‘#’
halive>     |
halive> 248 |                 #else
halive>     |                 ^
halive> 

--  While building package halive-0.1.6 using:
      /home/anton/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

This very compilation error is also the reason that `halive` is marked as broken on `nixpkgs`:


```
nix-repl> :b haskellPackages.halive
builder for '/nix/store/mk13jiivs3a080zb085i8r7ld1y3rg7x-halive-0.1.6.drv' failed with exit code 1; last 10 log lines:
  [2 of 8] Compiling Halive.Concurrent ( src/Halive/Concurrent.hs, dist/build/Halive/Concurrent.o )
  [3 of 8] Compiling Halive.FileListener ( src/Halive/FileListener.hs, dist/build/Halive/FileListener.o )
  [4 of 8] Compiling Halive.FindPackageDBs ( src/Halive/FindPackageDBs.hs, dist/build/Halive/FindPackageDBs.o )
  [5 of 8] Compiling Halive.SubHalive ( src/Halive/SubHalive.hs, dist/build/Halive/SubHalive.o )
  
  src/Halive/SubHalive.hs:248:17: error: parse error on input ‘#’
      |
  248 |                 #else
      |                 ^
  [8 of 8] Compiling Halive.Utils     ( src/Halive/Utils.hs, dist/build/Halive/Utils.o )
[0 built (1 failed), 59 copied (702.4 MiB), 35.6 MiB DL]
error: build of '/nix/store/mk13jiivs3a080zb085i8r7ld1y3rg7x-halive-0.1.6.drv' failed
```

--- a/demo/Window.hs
+++ b/demo/Window.hs
@@ -19,7 +19,7 @@ createGLWindow windowName = do
         -- , InitJoystick , InitGameController
         ]
     window <- createWindow windowName defaultWindow
-        { windowOpenGL = Just $ defaultOpenGL
+        { windowGraphicsContext = OpenGLContext $ defaultOpenGL
             { glProfile = Core Normal 4 1
             }
         , windowHighDPI = True

--- a/src/Halive/SubHalive.hs
+++ b/src/Halive/SubHalive.hs
@@ -243,11 +243,11 @@ recompileExpressionsInFile fileName mFileContents expressions =
                 -- Get the dependencies of the main target (and update the session with them)
                 graph <- depanal [] False
 
-                #if __GLASGOW_HASKELL__ >= 804
+#if __GLASGOW_HASKELL__ >= 804
                 let modSummaries = mgModSummaries graph
-                #else
+#else
                 let modSummaries = graph
-                #endif
+#endif
 
                 -- Load the dependencies of the main target
                 setContext

--- a/test/TestGHC.hs
+++ b/test/TestGHC.hs
@@ -88,11 +88,11 @@ main = withGHC $ do
             -- -- We must parse and typecheck modules before they'll be available for usage
             -- forM_ graph (typecheckModule <=< parseModule)
 
-            #if __GLASGOW_HASKELL__ >= 804
+#if __GLASGOW_HASKELL__ >= 804
             let modSummaries = mgModSummaries graph
-            #else
+#else
             let modSummaries = graph
-            #endif
+#endif
 
             -- Load the dependencies of the main target
             setContext

